merkle/sparse_merkle_tree.go

```diff
@@ -100,14 +100,15 @@ type subtreeWriter struct {
 // getOrCreateChildSubtree returns, or creates and returns, a subtree for the
 // specified childPrefix.
 func (s *subtreeWriter) getOrCreateChildSubtree(childPrefix []byte) (Subtree, error) {
-	childPrefixStr := string(childPrefix)
+	cp := append(make([]byte, 0, len(childPrefix)), childPrefix...)
```

Comment:
To be honest, I don't know - I added a bunch of copies trying to debug issues - I suspect that most (possibly all) of them will turn out to be unnecessary, but I'll go back and figure that out for sure once I'm back.  I'll pop another TODO in here to mark the copy.